### ISSUE TITLE
Video file name serialization doesn't work

### ISSUE THREAD

**Issue by AceWhatever** [axcore/tartube#568]
This is similar to [this currently unsolved issue](https://github.com/axcore/tartube/issues/527). Basically I'm trying to download multiple videos but alot of the sites I go to don't output a unique filename no matter what combination of factors I use in the renaming template (ie. no title, no ID and the autonumbering doesn't work). I don't know if this was done deliberately to prevent people from downloading consecutive videos quickly.

Regarding that last point, Tartube's default autonumbering is done using playlist_index. However the sites I'm downloading from do not have playlists implemented in anyway so the output is always "NA" for that.

So instead I tried using video_autonumber & autonumber. But that also doesn't work and results in Tartube giving a "file already exists" error after the first video is downloaded. The reason for this according to the yt-dlp Discord is "tartube is opening a new yt-dlp for each URL. We have no way to know what number to give." Which to my understanding means that autonumber can only increment the filenames in the existing session as long as it doesn't detect a file with the same name. This happens in both the regular video folders tab and the classic tab.

I know this sounds like more like a yt-dlp problem than Tartube, but hear me out. Assuming there isn't already a way to fix this problem or Tartube can't be changed to use one instance of yt-dlp for all current links, isn't there a way for the renaming to be done from the video folder tab? The way I see it, when adding multiple links, Tartube is adding them to a folder and indexing them in some way. Can't there be an option to at least add the link's numerical instance to the filename AFTER yt-dlp finishes downloading?

**Comment by axcore** [comment]
When the video download finishes, there is a delay (maybe a second, maybe two seconds) before Tartube can detect the downloaded file. If Tartube tries to rename the file, there will be another delay. These delays are caused by the operating system itself, not by Tartube or yt-dlp.

So by the time Tartube has finished the renaming process, yt-dlp will already have decided not to download other videos from the playlist, generating a "file already exists" error.

In short, I can't think of a way to resolve your problem.

**Comment by AceWhatever** [comment]
Can't Tartube be setup so that if I add say 5 videos to the folders tab and que them up for download, they get added as 5 separate download sessions instead of 1? That way yt-dlp's filename output options can be ignored by having Tartube rename the downloaded file with its own download rules before starting the next download so yt-dlp will never encounter the existing file error.

**Comment by axcore** [comment]
>Can't Tartube be setup so that if I add say 5 videos to the folders tab and que them up for download, they get added as 5 separate download sessions instead of 1?

Yes, that would work, but only when users remember to check all the videos first, and also remember to reduce the number of simultaneous downloads to one, and also remember that they have to select the videos and right-click to download them. And if a user forgets to do any of these things, or if they don't know how to do them, the downloads will fail.

Which is why my preferred solution would be for yt-dlp to generate a random number instead of NA for the video's ID, or something like that, which will work in all cases.
